developer/src/kmc-ldml/src/compiler/messages.ts

```diff
@@ -1,4 +1,5 @@
-import { util, CompilerErrorNamespace, CompilerErrorSeverity, CompilerMessageSpec as m, CompilerMessageDef as def } from "@keymanapp/common-types";
+import { CompilerErrorNamespace, CompilerErrorSeverity, CompilerMessageSpec as m, CompilerMessageDef as def } from "@keymanapp/common-types";
+import * as util from '@keymanapp/common-types/utils';
```

Comment:
We shouldn't be using deep imports between the kmc modules -- we are working towards well-defined API surfaces for each module, and this is a definite step backwards. Can we please only use top-level imports.

(Doesn't look like eslint has a [good rule](https://github.com/import-js/eslint-plugin-import/issues/1430) for this at present?)

Reply:
Context: https://github.com/keymanapp/keyman/pull/11881#discussion_r1687319082

Without this, KMW will grow approximately 4 KB in filesize.

Reply:
I saw the context. I am asking that we don't fix the growth issue by introducing a new problem.

We do _not_ need to use this pattern. We can just export the two functions that KMW needs from common-types/main.ts.

Reply:
> We do not need to use this pattern. We can just export the two functions that KMW needs from common-types/main.ts.

I do agree with that; it does feel like the better approach.

Reply:
Done

Reply:
ok, the reason I did it this way was to keep the size of `keymanweb.js` the same. To achieve this I moved the constants to `util/consts.ts`. These didn't get exported in `main.ts`, instead I exported `@keymanapp/common-types/utils`. Just exporting the two functions that KMW needs in addition to `*` didn't help with the size.

So I think we have two options:
- export the consts in `main.ts` again and live with the increased size,
- create a completely separate module for `common-types/utils`.

@mcdurdin @jahorton What do you think? If the second option, what should it be named?

Reply:
It's fine if the constants are the only "extras" along for the ride; their file-size contribution is minimal.  This is what I see on my local build - the _constants_ are pulled in, but they're just four extra lines, and the longest part of each is a variable name that can be substantially minified.

Here's the minified form of what we currently get:

```js
var dQ=55296,gQ=56319,uQ=56320,IQ=57343;function vs(Q){return Q>=dQ&&Q<=gQ}l(vs,"Uni_IsSurrogate1");function Ws(Q){return Q>=uQ&&Q<=IQ}l(Ws,"Uni_IsSurrogate2")
```

That's a total of... 40 characters.  It'd be lovely if we didn't have them, but it's not worth hyper-optimizing.

Reply:
So, to give a concise answer, I pick option 3:  neither.  Leave it as you currently have it.

Reply:
I'm not seeing the size issue at present -- I think @jahorton's final [comment above](https://github.com/keymanapp/keyman/pull/11881/files/02170dd0625763c5e3f4a89f19e26ff9dc8a654f#r1696162301) is probably the best answer (and should resolve the current build failures I hope!)

Reply:
yes, that works, thanks! Strange that it makes a difference whether or not the consts are defined and exported in the same file...